operator/pkg/util/yaml.go

```diff
@@ -65,13 +67,26 @@ func MarshalWithJSONPB(val proto.Message) (string, error) {
 	return string(yb), nil
 }
 
+var (
+	// stdErrMu is used to temporarily divert stderr to silence a bogus warning from external package.
+	stdErrMu sync.Mutex
+)
+
 // UnmarshalWithJSONPB unmarshals y into out using gogo jsonpb (required for many proto defined structs).
 func UnmarshalWithJSONPB(y string, out proto.Message, allowUnknownField bool) error {
 	jb, err := yaml.YAMLToJSON([]byte(y))
 	if err != nil {
 		return err
 	}
 
+	// Silence stray output to stderr in jsonpb package (issues/22261).
+	stdErrMu.Lock()
+	defer stdErrMu.Unlock()
+	stderr := os.Stderr
+	defer func() { os.Stderr = stderr }()
+	_, w, _ := os.Pipe()
```

Comment:
```suggestion
	stderr := os.Stderr
	_, w, _ := os.Pipe()
	defer w.Close()
	defer func() { os.Stderr = stderr }()
```

Reply:
something like that. Not sure it addresses any race TBH. But I think it clarifies the flow of cleanup.

Overall, I'm not a huge fan of intercepting os.Stderr for the whole process to address one bogus output in a single function. Are we sure no other routine/threads run concurrently and won't be silenced by this?

Reply:
Yeah, we cannot be sure of that. This was a solution of last resort because I've exhausted other reasonable options e.g. messing around with the tags in the generated structs to avoid this error.